ONSdigital/dp-dataset-api
# Create /population-types endpoint

### What

Added a new /population-types endpoint which exposes a list of population types available to query datasets against. Acceptance criteria is available here: https://trello.com/c/Tu3nq8kp/5481-create-new-get-census-endpoint-in-dp-dataset-api

### How to review

Code:
1. Verify unit and component tests are suitable and comprehensive enough
2. General code suggestions welcome

Manual test:
1. Run up docker compose stack specified here: https://github.com/ONSdigital/dp-compose/tree/main/cantabular-import
2. Stop the dp-dataset-api which starts as part of that stack
3. `git checkout feature/create-census-root-endpoint; git pull` for this repo
4. `ENABLE_PRIVATE_ENDPOINTS=false; make debug` to start the API without requiring authentication
5. curl or browse to `http://localhost:22000/population-types` and verify that content (and content type) similar to the below are returned (but note that the path is now /population-types NOT /census):

![image](https://user-images.githubusercontent.com/910448/151407835-b8d69526-ea4a-4d1b-ae09-06517e5272c5.png)


### Who can review

gophers (preferably having some knowledge of the census import journey)
